fix: alter mediation recipient websocket transport priority

JamesKEbert opened the pull request:
Alters the mediationRecipient to properly indicate the priority for using Websockets when doing implicit mediation. 
Updated `@aries-framework/react-native` to include buffer as a peer dependency to allow for processing and sending of messages via websockets in React Native, updated docs to reflect this.

@burdettadam assisted on this as well :)

TimoGlastra commented on packages/react-native/package.json:
```diff
@@ -28,19 +28,20 @@
+    "typescript": "~4.3.0"
   },
   "peerDependencies": {
+    "buffer": "^6.0.3",
```
`buffer` is already a dependency of core and we use it in other places. In `WsOutboundTransport` you can do: 

```
import { Buffer } from '../utils/buffer'
```

JamesKEbert commented on packages/core/src/modules/routing/RecipientModule.ts:
```diff
@@ -173,15 +181,15 @@ export class RecipientModule {
     const outboundMessage = createOutboundMessage(connection, message)
-    await this.messageSender.sendMessage(outboundMessage)
+
+    await this.sendMessage(outboundMessage)
```
I agree, I think that we'll want to adjust this in the future, especially with pickup-protocol v2.

JamesKEbert commented:
I've fixed a thing or two as requested, as well as fixed the failing tests. I added a try catch in the mediation websocket opener to catch failed attempts at opening the WS (motivated by iOS network errors). 
Let me know if there's anything else needed to change here @TimoGlastra or others.

TimoGlastra commented on packages/core/src/modules/routing/RecipientModule.ts:
```diff
@@ -85,17 +103,21 @@ export class RecipientModule {
+          // keepAlive: true,
+        },
+      })
+    } catch (error) {
+      this.logger.warn('Unable to open websocket connection to mediator', { error })
+    }
```
I'm not sure we want to swallow this error?

JamesKEbert replied:
The issue that I ran into is that it will throw an error if you don't have internet access, etc. Which means that if you throw it, it won't be caught and will not be able to reconnect properly. As far as I saw at least. Happy to go about this a different way in the future if desired.

TimoGlastra reviewed (approved):
One comment, but overall LGTM
